Handle new guava version (vendor class)

timja opened the pull request:
related to https://github.com/jenkinsci/bom/pull/423

cc @basil @dwnusbaum @car-roll @jglick @bitwiseman

jglick commented on src/main/java/jenkins/scm/api/InternetDomainName.java:
```diff
@@ -0,0 +1,648 @@
+ * the License.
+ */
+
+package jenkins.scm.api;
```
Do not put this in the API package.

timja replied:
moved

jglick commented:
merge with `master`

car-roll commented:
@bitwiseman do you have merge permissions?

timja commented:
@car-roll I fixed your permissions,

I've moved away from vendoring in here and used reflection, there was a lot of classes.
We could shade it if required: https://github.com/google/guava/wiki/UseGuavaInYourBuild#what-if-i-want-to-use-beta-apis-from-a-library-that-people-use-as-a-dependency

FTR it's still not out of beta in latest master: https://github.com/google/guava/issues/3423

timja commented:
@dwnusbaum

timja commented:
could someone merge and release?

basil commented:
FYI this still needs a release I think.
